### ISSUE TITLE
Add support for IP_ADDRESS:PORT format in XFF headers in WC_Geolocation::get_ip_address()

### ISSUE THREAD

**Issue by annemirasol** [woothemes/woocommerce#52390]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

As reported by a merchant, Amazon EC2 load balancers have a "Client port preservation" option, which leads to `$_SERVER['HTTP_X_FORWARDED_FOR']` to have a `IP_ADDRESS:PORT` format. `WC_Geolocation::get_ip_address()`'s current implementation rejects this format, which results in the method returning an empty string.

From [the X-Forwarded-For Wikipedia page](https://en.wikipedia.org/wiki/X-Forwarded-For):

> Because the X-Forwarded-For header is not formally standardized, some variations to the IP address format exist. For example, some implementations include the port number of clients...

### Expected behavior

I expected `WC_Geolocation::get_ip_address()` to be able to extract the IP Address from the `IP_ADDRESS:PORT` format.

### Actual behavior

If `X-Forwarded-For` has the `IP_ADDRESS:PORT` format, `WC_Geolocation::get_ip_address()` returns an empty string.

### Steps to reproduce

Simulation inside `wp shell`:

```
wp> unset($_SERVER['HTTP_X_REAL_IP']);
wp> $_SERVER['HTTP_X_FORWARDED_FOR'] = "10.0.0.1:80";
=> string(11) "10.0.0.1:80"
wp> echo 'IP Address: ' . WC_Geolocation::get_ip_address();
IP Address:
wp> $_SERVER['HTTP_X_FORWARDED_FOR'] = "10.0.0.1";
=> string(8) "10.0.0.1"
wp> echo 'IP Address: ' . WC_Geolocation::get_ip_address();
IP Address: 10.0.0.1
```

### WordPress Environment

WordPress 6.6.2
WooCommerce 9.3.3

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by gridig** [comment]
Thank you @annemirasol for opening this issue, for us it has caused multiple orders to fail with error “Payment failed: Invalid IP address. Order status changed from Pending payment to Failed.”

With "Client port preservation" off we're not seeing the error anymore, however correct IP extraction would be expected.

**Comment by mohitgusain** [comment]
@gridig @rodelgc @Konamiman, it would be really helpful if we could get a timeline for when this can be fixed.

**Comment by Konamiman** [comment]
Here's a PR: https://github.com/woocommerce/woocommerce/pull/52551

**Comment by mohitgusain** [comment]
@Konamiman @naman03malhotra It's been two weeks since the fix was added to the issue, but there has been no update since then. Could we be a bit more active on this, plan for the release, and include this update in the next plugin version?

**Comment by Konamiman** [comment]
@mohitgusain Sorry for the delay. The fix is merged and will land in WooCommerce 9.6.
